cmd/deploytargetconfig.go

```diff
@@ -48,31 +51,31 @@ var addDeployTargetConfigCmd = &cobra.Command{
 			return fmt.Errorf("Missing arguments: branches is a required flag")
 		}
 		current := lagoonCLIConfig.Current
-		lc := client.New(
+		token := lagoonCLIConfig.Lagoons[current].Token
+		lc := lclient.New(
 			lagoonCLIConfig.Lagoons[current].GraphQL,
-			lagoonCLIConfig.Lagoons[current].Token,
-			lagoonCLIConfig.Lagoons[current].Version,
 			lagoonCLIVersion,
+			&token,
 			debug)
-		project, err := lagoon.GetMinimalProjectByName(context.TODO(), cmdProjectName, lc)
+		project, err := l.GetMinimalProjectByName(context.TODO(), cmdProjectName, lc)
 		if err != nil {
 			return err
 		}
-		addDeployTargetConfig := &schema.AddDeployTargetConfigInput{
+		addDeployTargetConfig := &s.AddDeployTargetConfigInput{
 			Project: uint(project.ID),
 			Weight:  weight,
 		}
 		if branches != "" {
 			addDeployTargetConfig.Branches = branches
 		}
-		if branches != "" {
+		if pullrequests != "" {
 			addDeployTargetConfig.Pullrequests = pullrequests
 		}
 		if deploytarget != 0 {
 			addDeployTargetConfig.DeployTarget = deploytarget
 		}
```

Comment:
this needs an `else` statement that returns an error saying that a deploytarget id must be provided

Reply:
Updated to check for the presence of deploytarget id & project